src/librustc_codegen_ssa/base.rs

```diff
@@ -508,6 +509,48 @@ fn get_argc_argv<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
 
 pub const CODEGEN_WORKER_ID: usize = ::std::usize::MAX;
 
+/// Check for the `#[rustc_error]` annotation, which forces an
+/// error in codegen. This is used to write compile-fail tests
+/// that actually test that compilation succeeds without
+/// reporting an error.
+pub fn check_for_rustc_errors_attr(tcx: TyCtxt<'_>) {
```

Comment:
Maybe this should be in `rustc_driver`? Kind of weird to see it attached to codegen.

Reply:
It's only ever called from the beginning of `codegen_crate` (line 560 below)

Reply:
You can move the call to rustc_driver just before the call of codegen_crate.